src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/identifier_control_form.tsx

```diff
@@ -98,35 +117,48 @@ export function FieldControlForm({
   const isControlInEditMode = useMemo(() => !!initialState, [initialState]);
 
   useEffect(() => {
-    if (!availableFieldsOptions.length) {
-      const queryForFields = getQueryForFields(queryString, cursorPosition);
-      getESQLQueryColumnsRaw({
-        esqlQuery: queryForFields,
-        search,
-      }).then((columns) => {
-        setAvailableFieldsOptions(
-          columns.map((col) => {
-            return {
-              label: col.name,
-              key: col.name,
-              'data-test-subj': col.name,
-            };
-          })
-        );
-      });
+    if (!availableIdentifiersOptions.length) {
```

Comment:
nit: To indicate that we proceed only when the options are empty. Also, reduced indentation. 
```suggestion
    if (availableIdentifiersOptions.length > 0) return;
```

Reply:
Yo https://github.com/elastic/kibana/commit/7bb3bb14142baf4fbd9820d7feb558c6a292742a